src/mtcp/mtcp_restart.c

```diff
@@ -692,6 +693,26 @@ compute_vdso_vvar_addr(RestoreInfo *rinfo)
   mtcp_sys_close(mapsfd);
 }
 
+NO_OPTIMIZE
+static void
+unmap_one_memory_area(Area *area, int mapsfd)
+{
+  int mtcp_sys_errno;
+  DPRINTF("***INFO: munmapping (%p..%p)\n", area->addr, area->endAddr);
+  if (mtcp_sys_munmap(area->addr, area->size) == -1) {
+    MTCP_PRINTF("***WARNING: %s(%x): munmap(%p, %d) failed; errno: %d\n",
+                area->name, area->flags, area->addr, area->size,
+                mtcp_sys_errno);
+    mtcp_abort();
+  }
+
+  // Since we just unmapped a region, the size and contents of the
+  // /proc/self/maps file has changed. We should rewind and reread this file to
+  // ensure that we don't miss any regions.
```

Comment:
How could we miss a region?  We're calling unmap_one_memory_area in order on the areas.
Is this simply "defensive programming", or is there a scenario where we would miss something?  We're not closing and re-opening mapsfd -- so, lseek to offset 0 might simply show us the same stale data.